### PR TITLE
Improved navigation in source files

### PR DESCRIPTION
**Pull request by VlachJosef**
Improvements in RichPresentationCompiler. Method `symbolAt(pos:
Position)` now copy behaviour from Eclipse even more than before.

If point now allow jump to method apply, or some value, we will jump
to value, since it is more reasonable alternative.

### CODE REVIEW

File: core/src/main/scala/org/ensime/core/RichPresentationCompiler.scala [hunk 1]
```diff
@@ -478,32 +478,14 @@ class RichPresentationCompiler(
           }
         case Annotated(atp, _) =>
           List(atp.symbol)
-        case st: SymTree =>
+        case ap @ Select(qualifier, nme.apply) =>
+          // If we would like to give user choice if to go to method apply or value
+          // like Eclipse is doing we would need to return:
+          // List(qualifier.symbol, ap.symbol)
+          List(qualifier.symbol)
+        case st if st.symbol ne null =>
```
Comment (VlachJosef) [inline core/src/main/scala/org/ensime/core/RichPresentationCompiler.scala]: This is taken from [scala-ide ](https://github.com/scala-ide/scala-ide/blob/b2cb96d3d6735a4eb243a35b09e7835c8f8f8e18/org.scala-ide.sdt.core/src/org/scalaide/core/internal/hyperlink/ScalaDeclarationHyperlinkComputer.scala#L54)

File: core/src/main/scala/org/ensime/core/RichPresentationCompiler.scala [hunk 2]
```diff
@@ -478,32 +478,14 @@ class RichPresentationCompiler(
           }
         case Annotated(atp, _) =>
           List(atp.symbol)
-        case st: SymTree =>
+        case ap @ Select(qualifier, nme.apply) =>
```
Comment (VlachJosef) [inline core/src/main/scala/org/ensime/core/RichPresentationCompiler.scala]: This is taken from [scala-ide ](https://github.com/scala-ide/scala-ide/blob/b2cb96d3d6735a4eb243a35b09e7835c8f8f8e18/org.scala-ide.sdt.core/src/org/scalaide/core/internal/hyperlink/ScalaDeclarationHyperlinkComputer.scala#L53)

File: testing/simple/src/main/scala/org/example/Foo.scala [hunk 3]
```diff
@@ -18,15 +18,10 @@ object Foo extends App {
...
+  fn("foobar")
 }
 
 // for SearchServiceSpec
 case class CaseClassWithCamelCaseName()
 case class Bloo()
 case object Blue
-
-object Baz {
-  val fn: String => Int = str => str.size
```
Comment (fommil) [inline testing/simple/src/main/scala/org/example/Foo.scala]: I assume we were never testing this anyway

**Comment by fommil** [comment]
great stuff!

File: testing/simple/src/main/scala/org/example/Foo.scala (hunk 3 above)
Comment (VlachJosef) [inline testing/simple/src/main/scala/org/example/Foo.scala]: This was replaced by [this](https://github.com/ensime/ensime-server/blob/bc4ad93e2fc533f11ead04a66c6d7d476862dd5b/testing/simple/src/main/scala/org/example/Foo.scala#L19-L21).

**Pull request merged**